src/plugins/data/public/search/search_service.ts

```diff
@@ -135,11 +135,11 @@ export class SearchService implements Plugin<ISearchSetup, ISearchStart> {
 
     const aggTypesStart = this.aggTypesRegistry.start();
 
-    const search: ISearchGeneric = (request, options, strategyName) => {
+    const search: ISearchGeneric = (request, options) => {
       const { search: defaultSearch } = this.getSearchStrategy(
-        strategyName || DEFAULT_SEARCH_STRATEGY
+        options?.strategy || DEFAULT_SEARCH_STRATEGY
       );
-      return this.searchInterceptor.search(defaultSearch as any, request, options);
+      return this.searchInterceptor.search(defaultSearch, request, options);
```

Comment:
nit: in my opinion `defaultSearch` is not a very good name. Maybe we can rename it to `concreteSearch` or `specificSearch`.

Or do something like: 
```
const searchStrategy = this.getSearchStrategy(options?.strategy || DEFAULT_SEARCH_STRATEGY);
return this.searchInterceptor.search(searchStrategy.search, request, options);
```